developers/weaviate/client-libraries/typescript/v2_v3_migration.md

```diff
@@ -0,0 +1,283 @@
+    language="py"
+  />
+
+</TabItem>
+<TabItem value="query" label="NearText query">
+
+  <FilteredTextBlock
+    text={QuickStartCode}
+    startMarker="# NearTextExample"
+    endMarker="# END NearTextExample"
+    language="py"
+  />
+
+</TabItem>
+</Tabs>
+
+In both of these examples, you can see how the helper classes and methods abstract away the need for manual JSONs or strings.
+
+### Interaction with collections
+
+Interacting with the `client` object for CRUD and search operations have been replaced with the use of collection objects.
+
+This conveniently removes the need to specify the collection for each operation, and reduces potential for errors.
+
+import ManageDataCode from '!!raw-loader!/_includes/code/howto/manage-data.read.py';
+import ManageDataCodeV3 from '!!raw-loader!/_includes/code/howto/manage-data.read-v3.py';
+
+<Tabs groupId="languages">
+  <TabItem value="py" label="Python (v4)">
+    <FilteredTextBlock
+      text={ManageDataCode}
+      startMarker="# ReadObject START"
+      endMarker="# ReadObject END"
+      language="py"
+    />
+  </TabItem>
+
+  <TabItem value="py3" label="Python (v3)">
+    <FilteredTextBlock
+      text={ManageDataCodeV3}
+      startMarker="# ReadObject START"
+      endMarker="# ReadObject END"
+      language="py"
+    />
+  </TabItem>
+</Tabs>
+
+Note here that the collection object can be re-used throughout the codebase.
+
+### Collection creation from JSON
+
+You can still create a collection from a JSON definition. This may be a useful way to migrate your existing data, for example. You could [fetch an existing definition](../../manage-data/collections.mdx#read-a-single-collection-definition) and then use it to create a new collection.
+
+<FilteredTextBlock
+  text={PythonCode}
+  startMarker="# START CreateCollectionFromJSON"
+  endMarker="# END CreateCollectionFromJSON"
+  language="py"
+/>
+
+### Removal of builder patterns
+
+The builder patterns for constructing queries as been removed, as they could be confusing and potentially lead to invalid queries.
+
+In `v4`, queries are constructed using specific methods and its parameters.
+
+import SearchSimilarityCode from '!!raw-loader!/_includes/code/howto/search.similarity.py';
+import SearchSimilarityCodeV3 from '!!raw-loader!/_includes/code/howto/search.similarity-v3.py';
+
+<Tabs groupId="languages">
+  <TabItem value="py" label="Python (v4)">
+    <FilteredTextBlock
+      text={SearchSimilarityCode}
+      startMarker="# GetNearTextPython"
+      endMarker="# END GetNearTextPython"
+      language="python"
+    />
+  </TabItem>
+
+  <TabItem value="py3" label="Python (v3)">
+    <FilteredTextBlock
+      text={SearchSimilarityCodeV3}
+      startMarker="# GetNearTextPython"
+      endMarker="# END GetNearTextPython"
+      language="python"
+    />
+  </TabItem>
+</Tabs>
+
+This makes it easier to understand and use. Additionally, some parameters typed (e.g. `MetadataQuery`) which makes it easier to use and reduces errors.
```

Comment:
Consider:

The new syntax is easier to understand and use. Some parameters, like `MetadataQuery`, also have types. This reduces errors.

Reply:
Updates moved to other branches